slideactionview/src/main/java/me/jfenn/slideactionview/SlideActionView.java

```diff
@@ -1,98 +1,107 @@
+public class SlideActionView extends Component implements Component.DrawTask, Component.TouchEventListener {
+    private float mx = -1;
     private AnimatedFloat selected;
     private Map<Float, AnimatedFloat> ripples;
-
     private int handleRadius;
     private int expandedHandleRadius;
     private int selectionRadius;
     private int rippleRadius;
-
     private Paint normalPaint;
     private Paint outlinePaint;
     private Paint bitmapPaint;
-
-    private Bitmap leftImage, rightImage;
-
+    private PixelMap leftImage;
+    private PixelMap rightImage;
     private SlideActionListener listener;
 
+    /**
+     * SlideActionView Constructor.
+     *
+     *  @param context - context for SlideActionView constructor
+     *
+     */
     public SlideActionView(Context context) {
-        super(context);
+        this(context, null);
         init();
+        addDrawTask(this::onDraw);
     }
 
-    public SlideActionView(Context context, @Nullable AttributeSet attrs) {
-        super(context, attrs);
-        init();
-    }
-
-    public SlideActionView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
-        super(context, attrs, defStyleAttr);
+    /**
+     * SlideActionView Constructor.
+     *
+     *  @param context - context for SlideActionView constructor
+     *  @param attrSet - attributes
+     *
+     */
+    public SlideActionView(Context context, AttrSet attrSet) {
+        super(context, attrSet, 0);
         init();
+        addDrawTask(this::onDraw);
     }
 
-    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
-    public SlideActionView(Context context, @Nullable AttributeSet attrs, int defStyleAttr, int defStyleRes) {
-        super(context, attrs, defStyleAttr, defStyleRes);
+    /**
+     * SlideActionView Constructor.
+     *
+     *  @param context - context for SlideActionView constructor
+     *  @param attrSet - attributes
+     *  @param defStyleAttr - defStyle attribute
+     *
+     */
+    public SlideActionView(Context context, AttrSet attrSet, int defStyleAttr) {
+        super(context, attrSet, defStyleAttr);
         init();
+        addDrawTask(this::onDraw);
     }
 
     private void init() {
-        handleRadius = DimenUtils.dpToPx(12);
-        expandedHandleRadius = DimenUtils.dpToPx(32);
-        selectionRadius = DimenUtils.dpToPx(42);
-        rippleRadius = DimenUtils.dpToPx(140);
-
+        handleRadius = DimenUtils.dpToPx(getContext(), 12);
+        expandedHandleRadius = DimenUtils.dpToPx(getContext(), 32);
+        selectionRadius = DimenUtils.dpToPx(getContext(), 42);
+        rippleRadius = DimenUtils.dpToPx(getContext(), 140);
         selected = new AnimatedFloat(0);
         ripples = new HashMap<>();
-
         normalPaint = new Paint();
-        normalPaint.setStyle(Paint.Style.FILL);
-        normalPaint.setColor(Color.GRAY);
+        normalPaint.setStyle(Paint.Style.FILL_STYLE);
+        Color hmosColor = SlideActionView.changeParamToColor(Color.RED.getValue());
```

Comment:
Please keep the color codes similar to the library's base code..as it is hardcoded within the library. Applies to all the similar instances which are present below in the same file

Reply:
Done